### PR TITLE
boot: Make initrd_prepare() semantically equivalent to combine_initrds()

### PR DESCRIPTION
**Pull request by daandemeyer**
Currently, trying to boot images with type 1 entries generated by mkosi with qemu freezes in the kernel EFI stub. I'm not going to pretend I understand what's going on, but when I reported a similar problem with UKIs, the fix was to rework the code in combine_initrds() in the stub to behave like it does today. It seems that same fix was never applied to systemd-boot's combine_initrds() function, so let's do that now to fix the freezes I've been seeing trying to boot images with type 1 entries in qemu.

### CODE REVIEW

File: src/boot/efi/boot.c [hunk 1]
```diff
@@ -2464,31 +2464,60 @@ static EFI_STATUS initrd_prepare(
...
         if (entry->options) {
                 _cleanup_free_ char16_t *o = options;
                 options = xasprintf("%ls %ls", o, entry->options);
         }
 
         *ret_options = TAKE_PTR(options);
-        *ret_initrd = TAKE_PTR(initrd);
+        *ret_initrd_pages = pages;
         *ret_initrd_size = size;
+        pages.n_pages = 0;
```
Comment (yuwata) [inline src/boot/efi/boot.c]: `TAKE_STRUCT()` can be used?

**Review by yuwata — APPROVED** [review]
LGTM.

**Comment by poettering** [comment]
lgtm

**Review by bluca — APPROVED** [review]

**Comment by bluca** [comment]
these do not apply cleanly to v256-stable

**Comment by DaanDeMeyer** [comment]
First one doesn't matter and I already backported the second one myself

**Pull request merged**